eofconvention.fsx

```diff
@@ -0,0 +1,26 @@
+open System.IO
+
+let EolAtEof(path: FileInfo) = 
+
+    use streamReader = new StreamReader (path.FullName)
+    let filetext = streamReader.ReadToEnd()
+    
+    if filetext <> "" then
+        (filetext |> Seq.last = '\n')
+    else
+        true
+
+let NotPrefix (prefix: string) (mainStr: string) =
+    not (mainStr.StartsWith(prefix))
+
+let invalidFiles = 
+    Directory.GetFiles(".", "*.*", SearchOption.AllDirectories) 
+    |> Seq.filter (NotPrefix "./node_modules")
+    |> Seq.filter (NotPrefix "./.git")
+    |> Seq.map (fun pathStr -> FileInfo pathStr)
+    |> Seq.filter (fun fileInfo -> not (EolAtEof fileInfo))
+
+if invalidFiles |> Seq.length > 0 then
```

Comment:
The pipe |> operator is good for what you did in lines 16-21, a chain-like thing, but when we're talking about a single function with a single argument, it's just much shorter and readable to not use it

Reply:
What about line 25? Should I change that too?

Reply:
well line 25 deserves a different conversation :) because using a mutable variable there can be avoided

Reply:
I tried to fix it. But now it looks a bit strange